src/blosc2/blosc2_ext.pyx

```diff
@@ -1207,8 +1219,55 @@ cdef class SChunk:
             raise RuntimeError("Could not append the buffer")
         return rc
 
-    def fill_special(self, nitems, special_value):
-        return blosc2_schunk_fill_special(self.schunk, nitems, special_value, self.chunksize)
+    def fill_special(self, nitems, special_value, value):
+        if value is None:
+            return blosc2_schunk_fill_special(self.schunk, nitems, special_value, self.chunksize)
+
+        if nitems == 0:
+            return 0
+        if nitems * self.typesize / self.chunksize > INT_MAX:
+            raise RuntimeError("nitems is too large.  Try increasing the chunksize")
+        if self.nbytes > 0 or self.cbytes > 0:
+            raise RuntimeError("Filling with special values only works on empty SChunks")
+
+        array = np.array([value])
+        if array.dtype.itemsize != self.typesize:
+            if isinstance(value, int):
+                dtype = np.dtype('i'+ str(self.typesize))
+            elif isinstance(value, float):
+                dtype = np.dtype('f' + str(self.typesize))
+            else:
+                raise ValueError("value size in bytes must match with typesize")
+            array = np.array([value], dtype=dtype)
+        cdef Py_buffer *buf = <Py_buffer *> malloc(sizeof(Py_buffer))
+        PyObject_GetBuffer(array, buf, PyBUF_SIMPLE)
+
+        nchunks = nitems // self.chunkshape
+        cdef blosc2_schunk *c_schunk = <blosc2_schunk *> self.c_schunk
+        cdef blosc2_cparams *cparams = self.schunk.storage.cparams
+        chunksize = BLOSC_EXTENDED_HEADER_LENGTH + self.typesize
+        cdef void *chunk = malloc(chunksize)
+        get_chunk_repeatval(dereference(cparams), self.chunksize, chunk, chunksize, buf)
+
+        for i in range(nchunks):
+            if blosc2_schunk_append_chunk(self.schunk, <uint8_t *>chunk, True) < 0:
+                free(chunk)
+                PyBuffer_Release(buf)
+                free(buf)
+                raise RuntimeError("Error while appending the chunk")
+
+        last_nitems = nitems % self.chunkshape
```

Comment:
I'd call this variable `remainder`.

Reply:
Done in 296712f60e1f58169fe0f3d0766d30ff8d981e30.